src/identity/manager.rs

```diff
@@ -1109,6 +1159,36 @@ mod tests {
         assert!(sm.fetch_certificate(&id).await.is_ok());
     }
 
+    #[tokio::test(start_paused = true)]
+    async fn test_get_existing_cert_info_basic() {
+        let test = setup(1);
+        let id = identity("basic-test");
+        let info = test.secret_manager.worker.get_existing_cert_info(&id).await;
+        assert!(info.is_none());
+
+        // cleanup
+        test.tear_down().await;
+    }
+
+    #[tokio::test(start_paused = true)]
+    async fn test_certificate_retention_functionality() {
```

Comment:
I am not sure how this tests the retained functionality?

I was expecting to see a test like

```
fetch succeeds
fetch fails
we can still get the cert
```
?